main/lsp/requests/rename.cc

```diff
@@ -187,13 +187,25 @@ variant<JSONNullObject, unique_ptr<WorkspaceEdit>> Renamer::buildEdit() {
         return JSONNullObject();
     }
 
-    auto we = make_unique<WorkspaceEdit>();
+    UnorderedMap<string, vector<unique_ptr<TextEdit>>> tmpEdits;
     vector<unique_ptr<TextDocumentEdit>> textDocEdits;
+    // collect changes per file
     for (auto &item : edits) {
+        core::Loc loc = item.first;
+        string newsrc = item.second;
+        auto location = config.loc2Location(gs, loc);
+        ENFORCE(location != nullptr); // loc should always exist
+        if (location == nullptr) {
+            continue;
+        }
+        tmpEdits[location->uri].push_back(make_unique<TextEdit>(move(location->range), newsrc));
```

Comment:
```suggestion
        tmpEdits[location->uri].push_back(make_unique<TextEdit>(move(location->range), move(newsrc)));
```

No need to copy the string twice.